src/Controls/src/Core/Platform/AlertManager/AlertManager.Windows.cs

```diff
@@ -231,6 +241,17 @@ static async Task<bool> ShowAlert(ContentDialog alert)
 
 				return null;
 			}
+
+			bool PageIsInThisWindow(Page page)
+			{
+				var window = page?.Window;
+				var platformWindow = window?.MauiContext.GetPlatformWindow();
+
+				if (platformWindow?.GetHashCode() == Window.GetHashCode())
```

Comment:
Does this need to be a GetHashCode comparison? can't we just compare the two windows?

Reply:
It definitely _should not_ be a hash code comparison; there's nothing preventing the two windows from having the same hash code and displaying the alert in multiple windows.